packages/core/src/components/button/buttons.tsx

```diff
@@ -73,13 +87,27 @@ export const AnchorButtonFactory = React.createFactory(AnchorButton);
 function getButtonClasses(props: IButtonProps) {
     return classNames(
         Classes.BUTTON,
-        { [Classes.DISABLED]: props.disabled },
+        { [Classes.DISABLED]: isButtonDisabled(props) },
         Classes.iconClass(props.iconName),
         Classes.intentClass(props.intent),
         props.className,
     );
 }
 
+function isButtonDisabled(props: IButtonProps) {
+    return props.disabled === true || props.loading === true;
+}
+
+function buttonContentsVisibility(loading: boolean) {
+    return {visibility: loading ? "hidden" : "visible"};
+}
+
+function maybeRenderSpinner(loading: boolean) {
+    return loading != null && loading
+      ? <span className="pt-button-spinner-container"><Spinner className="pt-small" /></span>
```

Comment:
is there any way to re-use the `Spinner`'s root element as the `pt-button-spinner-container`? so we don't need to add yet another element.